slack_sdk/web/async_client.py

```diff
@@ -2244,6 +2244,107 @@ async def calls_update(
         )
         return await self.api_call("calls.update", http_verb="POST", params=kwargs)
 
+    async def canvases_create(
+        self,
+        *,
+        title: Optional[str] = None,
+        document_content: Dict[str, str],
+        **kwargs,
+    ) -> AsyncSlackResponse:
+        """Create Canvas for a user
+        https://api.slack.com/methods/canvases.create
+        """
+        kwargs.update({"title": title, "document_content": json.dumps(document_content)})
+        return await self.api_call("canvases.create", params=kwargs)
+
+    async def canvases_edit(
+        self,
+        *,
+        canvas_id: str,
+        changes: Sequence[Dict[str, Any]],
+        **kwargs,
+    ) -> AsyncSlackResponse:
+        """Update an existing canvas
+        https://api.slack.com/methods/canvases.edit
+        """
+        kwargs.update({"canvas_id": canvas_id, "changes": json.dumps(changes)})
+        return await self.api_call("canvases.edit", params=kwargs)
+
+    async def canvases_delete(
+        self,
+        *,
+        canvas_id: str,
+        **kwargs,
+    ) -> AsyncSlackResponse:
+        """Deletes a canvas
+        https://api.slack.com/methods/canvases.delete
+        """
+        kwargs.update({"canvas_id": canvas_id})
+        return await self.api_call("canvases.delete", params=kwargs)
+
+    async def canvases_access_set(
+        self,
+        *,
+        canvas_id: str,
+        access_level: str,
+        channel_ids: Optional[Union[Sequence[str], str]] = None,
+        user_ids: Optional[Union[Sequence[str], str]] = None,
+        **kwargs,
+    ) -> AsyncSlackResponse:
+        """Sets the access level to a canvas for specified entities
+        https://api.slack.com/methods/canvases.access.set
+        """
+        kwargs.update({"canvas_id": canvas_id, "access_level": access_level})
+        if channel_ids is not None:
+            if isinstance(channel_ids, (list, Tuple)):
+                kwargs.update({"channel_ids": ",".join(channel_ids)})
+            else:
+                kwargs.update({"channel_ids": channel_ids})
+        if user_ids is not None:
```

Comment:
Just a note as a possible enhancement in the future, I see that both client and async client handle this list serialization into comma-separated strings at the method level. Perhaps an enhancement could be to factor that out into a utility method? I see the same pattern is repeated a lot: 

```
➜ ack "list, Tuple"
client.py
289:            if isinstance(team_ids, (list, Tuple)):
344:        if isinstance(app_ids, (list, Tuple)):
403:        if isinstance(entity_ids, (list, Tuple)):
422:        if isinstance(entity_ids, (list, Tuple)):
442:        if isinstance(barriered_from_usergroup_ids, (list, Tuple)):
446:        if isinstance(restricted_subjects, (list, Tuple)):
477:        if isinstance(barriered_from_usergroup_ids, (list, Tuple)):
```

Reply:
yeah it should be good; we can do it in the future